fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/FullUnorderedScanExpression.java

```diff
@@ -134,7 +144,15 @@ public String toString() {
     @Nonnull
     @Override
     public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candidateExpression, @Nonnull final AliasMap aliasMap, @Nonnull final IdentityBiMap<Quantifier, PartialMatch> partialMatchMap) {
-        return exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
+        if (getClass() != candidateExpression.getClass()) {
+            return ImmutableList.of();
+        }
+        // if query doesnot contain candidate's indexes, the query cannot be subsumed by the candidate
+        if (getAccessHints().containsAll(((FullUnorderedScanExpression)candidateExpression).getAccessHints())) {
```

Comment:
rename `containsAll()` to something like `satisfies()`. We want to encapsulate what it does. In fact it's not a a straight up `constainsAll()`, it has special logic for empty which is required but kind of clashing with `constainsAll`